python/ffsim/states/slater.py

```diff
@@ -0,0 +1,326 @@
+            entries represent accepted or rejected moves respectively.
+        new_pos: A 2D Numpy array with the proposed new positions of the
+            electrons. Each row represents an electronic configuration.
+        old_pos: A 2D Numpy array with the previous positions of the
+            electrons. Each row represents an electronic configuration.
+
+    Returns:
+        A 2D Numpy array with the previous positions of the
+        electrons after accepting or rejecting the electron moves according to
+        the ``acceptance_vector``.
+
+
+    """
+    for i in range(new_pos.shape[0]):
+        if not acceptance_vector[i]:
+            new_pos[i, :] = old_pos[i, :]
+
+    return new_pos
+
+
+def _positions_to_fock(positions: np.ndarray, norb: int) -> np.ndarray:
+    """
+    Transforms electronic configurations defined by the position of the electrons
+    to the occupation representation.
+
+    Args:
+        positions: A Numpy array of positions of electrons. Each row corresponds
+            to an electronic configuration.
+        norb: number of orbitals.
+
+    Returns:
+        A 2Dimensional Numpy array of electronic configurations represented by
+        the orbital occupancy. Each row is an electronic configuration.
+    """
+
+    n_samples = positions.shape[0]
+    fock = np.zeros((n_samples, norb)).astype("int")
+    for i in range(n_samples):
+        fock[i, norb - 1 - positions[i, :]] = 1
+    return fock
+
+
+def _sample_spinless(
+    rdm: np.ndarray,
+    chain_length: int,
+    n_chains: int,
+    n_particles_to_move: int = 1,
+    seed: int | None = None,
+) -> np.ndarray:
+    """
+    Collect samples of electronic configurations from a Slater determinant defined
+    by its one-body reduced density matrix (RDM). The sampler uses the
+    Metropolis-Hastings method.
+
+    Args:
+        rdm: The one-body reduced density matrix that defines the Slater determinant
+            This is a single Numpy array specifying the 1-RDM of a spin-polarized
+            system.
+        chain_length: Length of each sampling chain.
+        n_chains: Number of chains to run. The total number of samples is given
+            by (``chain_length`` + 1) * ``n_chains``.
+        n_particles_to_move: At each step of the chain, the maximum number of
+            electrons that are moved into a different orbital.
+        seed: Seed for the random number generator.
+
+    Returns:
+        A 2Dimensional Numpy array with samples of electronic configurations.
+        Each row is a sample.
+    """
+
+    np.random.seed(seed)
+
+    norb = rdm.shape[0]
+
+    nelec = round(np.sum(np.diag(rdm)))
+
+    if nelec < n_particles_to_move:
+        raise ValueError(
+            "Number of electrons smaller than ``n_particles_to_move``: "
+            + "number of electrons ("
+            + str(nelec)
+            + ") and  ``n_particles_to_move`` ("
+            + str(n_particles_to_move)
+            + ")"
```

Comment:
The `+`s are unnecessary. Also, please use f-strings. Something like this:
```suggestion
            "Number of electrons smaller than ``n_particles_to_move``: "
            f"number of electrons ({nelec}) "
            f"and ``n_particles_to_move`` ({n_particles_to_move})."
```